### PR TITLE
interop-testing: fix interop_test client.

### LINKED ISSUE #3562: Many java cloud_to_cloud and cloud_to_prod interop tests failling: " No ServerStreamTracer found"
```
2017-10-09 16:53:03,682 START: cloud_to_cloud:node:csharpcoreclr_server:unimplemented_method
2017-10-09 16:53:03,686 Gradle is no longer run automatically. Make sure to run
'./gradlew installDist -PskipCodegen=true' or
'./gradlew :grpc-interop-testing:installDist -PskipCodegen=true' after any
changes. -PskipCodegen=true is optional, but requires less setup.
[jetty-alpn-agent] Using: alpn-boot-8.1.11.v20170118.jar
Running test status_code_and_message
Exception in thread "main" java.lang.AssertionError: No ServerStreamTracer found
	at io.grpc.testing.integration.AbstractInteropTest.assertServerMetrics(AbstractInteropTest.java:1828)
	at io.grpc.testing.integration.AbstractInteropTest.assertMetrics(AbstractInteropTest.java:1709)
	at io.grpc.testing.integration.AbstractInteropTest.assertMetrics(AbstractInteropTest.java:1717)
	at io.grpc.testing.integration.AbstractInteropTest.statusCodeAndMessage(AbstractInteropTest.java:1395)
	at io.grpc.testing.integration.TestServiceClient.runTest(TestServiceClient.java:282)
	at io.grpc.testing.integration.TestServiceClient.run(TestServiceClient.java:189)
	at io.grpc.testing.integration.TestServiceClient.main(TestServiceClient.java:66)

2017-10-09 16:53:03,686 TIMEOUT: cloud_to_cloud:java:java_server:status_code_and_message [pid=30665, time=180.4sec]
```

cloud_to_cloud:
https://sponge.corp.google.com/target?id=23de9377-477a-4cf9-876c-20df54eb5e83&target=github/grpc&searchFor=&show=ALL&sortBy=STATUS

cloud_to_prod:
https://sponge.corp.google.com/target?id=3fc281f5-0428-4d03-bc10-8fb9b3a7d035&target=github/grpc&searchFor=&show=ALL&sortBy=STATUS

### PR DESCRIPTION
**Pull request by zhangkun83**
Resolves #3562

### CODE REVIEW

File: interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java [hunk 1]
```diff
@@ -351,23 +353,25 @@ protected ManagedChannel createChannel() {
...
         }
-        return builder.build();
+        builder = okBuilder;
       }
+      io.grpc.internal.TestingAccessor.setStatsContextFactory(builder, getClientStatsFactory());
+      return builder.build();
     }
 
     @Override
-    protected boolean metricsExpected() {
```
Comment (ericgribkoff) [inline interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java]: It looks like metricsExpected() will now return true for the interop client - only serverInProcess() returning false is required now?

Comment (zhangkun83) [reply interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java]: Yes. serverInProcess() is the most precise control.

**Comment by ericgribkoff** [comment]
@zhangkun83 The Travis OSX build failed on this PR:
https://travis-ci.org/grpc/grpc-java/jobs/286166891

> io.grpc.testing.integration.InProcessTest > statusCodeAndMessage FAILED
    java.lang.AssertionError: expected:<UNKNOWN> but was:<CANCELLED>

I haven't seen this failure previously. Is this a known flake?

**Comment by zhangkun83** [comment]
Doesn't seem so. Can you file an issue to track it?

**Comment by ejona86** [comment]
That looks like another random flake I've seen from today (also in InProcessTest). Both I think are caused by 30 second timeout being triggered. I think that's because Travis was having trouble today. I decided to ignore the other one. Let's ignore this one too. If we still see something like it again tomorrow then let's create the issue.

**Comment by zhangkun83** [comment]
Seen this again today. Filed #3571

**Pull request merged**